### ISSUE TITLE
[DesignProposal]: EnsembleImageEmbeddingPipeline for multiple image models ensemble.

### ISSUE THREAD

**Issue by wxywb** [towhee-io/towhee#516]
### Background and Motivation

There is a lot of needs in ML community that practioners wants ensemble some SoTA models to solve their problems quickly. Providing the ability to ensemble as a tool can meet with their demands.

### Design
There is plenty of methods(boosting，stacking,  bagging) , and stacking could have the higher priority to implement.
user can have multiple choices to configure a specific stacking EnsembleImageEmbeddingPipeline. Since it has some 
states (model weights or blending coefficient), so we need a class to maintain its states.
```python
# ensemble construction
ensemble_pipeline = EnsembleImageEmbeddingPipeline(
# names should be image embedding operators existed in hub 
  models = [‘resnet-101’, 'swin-transformer-L','efficientnetB6'], 
  base_learner_postprocess = Identity(),
  aggregation = Concat(),
  meta_learner = Identity(),
  pretrainened = None,
  train_configs = training_configs)

# ensemble train (there are many chances in the training procedure to serialize the model weights, so no need to call save())
ensemble_pipeline.train()

# ensemble inference
emb = ensemble_pipeline(Image.open('test.png'))

```
**notes:** customized classes need to be implemented in a backward-able way to make whole ensemble pipeline trainable.

#### params:
**models**: ensemble models' names.
**base_learner_postprocess**: how to process individual model's (e.g. `Identity()`, `Normalization()`, `Weighted(1,3,5)`)
**aggregation**: how to aggregate  individual model's output.(e.g. `Average()`, `Concat()`)
**meta_learner**: meta-learner type(e.g. `FCNet()`, `Identity()`)
**train_configs**: training configs to construct trainer(datasets, loss, optimizer, learning rate scheduler, etc)
### Pros and Cons
**pros:** flexible, can combine the options(can be customized by users) to varies ensembles for varies tasks. 
**cons:** need to go through docs to checks which option is provided.
